api/src/main/java/org/opennms/integration/api/v1/detectors/ServiceDetectorFactory.java

```diff
@@ -0,0 +1,76 @@
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
+ *
+ * Copyright (C) 2018 The OpenNMS Group, Inc.
+ * OpenNMS(R) is Copyright (C) 1999-2018 The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is a registered trademark of The OpenNMS Group, Inc.
+ *
+ * OpenNMS(R) is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.integration.api.v1.detectors;
+
+import java.net.InetAddress;
+import java.util.Map;
+
+import org.opennms.integration.api.v1.annotations.Exposable;
+
+/**
+ *  Responsible for instantiating service detectors.
+ * @param <T> Class that implements {@link ServiceDetector}
+ */
+@Exposable
+public interface ServiceDetectorFactory<T extends ServiceDetector> {
+
+    /**
+     * Instantiates a new detector and set bean properties.
+     * One of the ways to set bean properties is by using Spring BeanWrapper.
+     * <pre>
+     * {@code
+     *         BeanWrapper wrapper = PropertyAccessorFactory.forBeanPropertyAccess(serviceDetector);
+     *         wrapper.setPropertyValues(properties);
+     * }
+     * </pre>
+     * Detectors are treated as protoypes meaning there will be a new instance of detector each time createDetector gets called.
+     * @param properties  are used to set properties on detector bean.
+     *
+     * @return a new @{@link ServiceDetector}.
+     */
+    T createDetector(Map<String, String> properties);
```

Comment:
Since now it is factory's responsibility to set bean properties, exposing it in Integration-API as well.  Implementations can decide how to set those properties.